core/commands/storage/upload/upload/inquiry.go

```diff
@@ -0,0 +1,196 @@
+package upload
+
+import (
+	"context"
+	"errors"
+	"github.com/TRON-US/go-btfs/core/hub"
+	"time"
+
+	cmds "github.com/TRON-US/go-btfs-cmds"
+	"github.com/TRON-US/go-btfs/core"
+	"github.com/TRON-US/go-btfs/core/commands/cmdenv"
+	uh "github.com/TRON-US/go-btfs/core/commands/storage/upload/helper"
+	"github.com/TRON-US/go-btfs/core/corehttp/remote"
+	guardpb "github.com/tron-us/go-btfs-common/protos/guard"
+	"github.com/tron-us/go-btfs-common/utils/grpc"
+	"github.com/tron-us/protobuf/proto"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+var StorageRenewInquiryCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "inquire hosts for the new duration and return the hosts signed contracts.",
+		ShortDescription: `
+Communicate with hosts to accept or reject the new contracts, return the signed the contracts if agree with the extended period, 
+or find new hosts to download and store the shards.`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("renter-pid", true, false, "Original renter peer ID."),
+		cmds.StringArg("file-hash", true, false, "Hash of file to upload."),
+		cmds.StringArg("entry-key", true, false, "Key for querying meta data from Guard."),
```

Comment:
the key will be passed from guard, guard will generate a uuid as a key to repair node,  so that repair node can use it to retrieve the filestorestatus without admin signature.